core/src/main/java/hudson/model/Queue.java

```diff
@@ -433,6 +441,22 @@ public synchronized void load() {
         }
     }
 
+    private long getMaxNextBuildNumber() {
+        long maxNextBuildNumber = 0;
+        Jenkins jenkins = Jenkins.getInstance();
+
+        if (jenkins != null) {
+            for (String jobName : jenkins.getJobNames()) {
+                TopLevelItem item = jenkins.getItem(jobName);
+                if (item instanceof Job) {
+                    maxNextBuildNumber = Math.max(maxNextBuildNumber, ((Job) item).getNextBuildNumber());
```

Comment:
If someone has more than 2^31-1 builds of a given job then the `nextBuildNumber` is going to wrap around and be screwed. What way are you saying it should be if this is wrong?